azext_edge/edge/providers/orchestration/resources/instances.py

```diff
@@ -50,28 +48,25 @@ def list(self, resource_group_name: Optional[str] = None) -> Iterable[dict]:
         return self.iotops_mgmt_client.instance.list_by_subscription()
 
     def _show_tree(self, instance: dict):
-        custom_location = self.get_associated_cl(instance)
-        if not custom_location:
-            logger.warning("Unable to process the resource tree.")
-            return
-
-        resource_id_container = parse_resource_id(custom_location["properties"]["hostResourceId"])
-
-        # Currently resource map will query cluster state upon init
-        # therefore we only use it when necessary to save cycles.
-        from ..resource_map import IoTOperationsResourceMap
-
+        resource_map = self.get_resource_map(instance)
         with self.console.status("Working..."):
-            resource_map = IoTOperationsResourceMap(
-                cmd=self.cmd,
-                cluster_name=resource_id_container.resource_name,
-                resource_group_name=resource_id_container.resource_group_name,
-            )
+            resource_map.refresh_resource_state()
         print(resource_map.build_tree(category_color="cyan"))
 
-    def get_associated_cl(self, instance: dict) -> dict:
-        return self.query(
-            QUERIES["get_cl_from_instance"].format(resource_id=instance["extendedLocation"]["name"]), first=True
+    def _get_associated_cl(self, instance: dict) -> dict:
```

Comment:
not sure if this should be abstracted since the assets also need the custom location...

Reply:
unless you expect us to create an instance of this class for assets/other rpsaas services

Reply:
What do you mean by this ? The instance has reference to the custom location Id. What do you need from the custom location to create an asset ?

Reply:
I need the custom location id (for the extended location) + all the cluster + custom location checks for asset/endpoint creation...